### PR TITLE
Simplify DSL generation

### PR DESCRIPTION
**Pull request by radtriste**
- Remove the per-repo seed jobs => only one seed job per branch
- Keep per-repo trigger jobs to be sure that it triggers the main seed job when jenkins folder is updated on each repo
- Refactor DSL models for more flexibility
  - Switch models from enum to class
  -  Automatic creation of folders, instead of static ones. Gives more flexibility to add new Folders

### CODE REVIEW

File: dsl/seed/jobs/Jenkinsfile.seed.branch [hunk 1]
```diff
@@ -19,69 +35,127 @@ pipeline {
...
 
-        stage('Launch repository seed jobs') {
+        stage('Generate jobs') {
             steps {
                 script {
-                    Map parallelJobs = [:]
-                    util.readBranchConfig().repositories.collect { it.name }.each { repository ->
-                        parallelJobs.put(repository, { launchRepositorySeedJob(repository) })
+                    all_repos.each { repository ->
+                        def envProps = getRepoConfigAsEnvProperties(repository)
```
Comment (Ginxo) [inline dsl/seed/jobs/Jenkinsfile.seed.branch:102]: your proposal is fine, just another POV to obtain the same. To remove the `envProps.put` lines and
```suggestion
                        def envProps = getRepoConfigAsEnvProperties(repository) + ['GIT_MAIN_BRANCH': getMainBranch(repository), 'REPO_NAME': "${repository}"]
```

see 
image

Comment (Ginxo) [reply dsl/seed/jobs/Jenkinsfile.seed.branch:102]: or you can even do and to remove `envProps.put` and `all_repos.each`
```suggestion
                        def envProps = getRepoConfigAsEnvProperties(repository) 
                                                 + ['GIT_MAIN_BRANCH': getMainBranch(repository), 'REPO_NAME': "${repository}"]
                                                 + all_repos.collectEntries { [util.generateEnvKey(it, 'jenkins_config_path'), getRepoConfig(it).git.jenkins_config_path] }
```
your choice

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Environment.groovy [hunk 2]
```diff
@@ -11,17 +11,51 @@ import org.kie.jenkins.jobdsl.Utils
 * Also an environment can be optional.
 * So an environment can be disabled and the method `isActiveClosure` should be initialized else it returns `true` by default.
 *
+*
 */
-enum Environment {
+class Environment {
```
Comment (Ginxo) [inline dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Environment.groovy:16]: I like your style :cupid:

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy [hunk 3]
```diff
@@ -167,26 +84,169 @@ enum Folder {
...
+    public static final Folder RELEASE = new Folder(
+        name: 'RELEASE',
+        jobType: JobType.RELEASE,
+        environment: Environment.DEFAULT,
+        defaultEnv: [
+            RELEASE: 'true'
+        ],
+    )
+
+    // public static final Folder UPDATE_VERSION = new Folder(
```
Comment (Ginxo) [inline dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy]: should these commented lines be removed?

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/JobType.groovy [hunk 4]
```diff
@@ -8,25 +8,36 @@ import org.kie.jenkins.jobdsl.Utils
...
+        name: 'PULLREQUEST',
+    )
+    public static final JobType NIGHTLY = new JobType(
+        name: 'NIGHTLY',
         optional: true,
-    ),
-    // UPDATE_VERSION,
-    RELEASE(
+    )
+    // public static final JobType UPDATE_VERSION  = new JobType (
```
Comment (Ginxo) [inline dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/JobType.groovy]: to remove?

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/JobType.groovy [hunk 5]
```diff
@@ -36,4 +47,26 @@ enum JobType {
     boolean isActive(def script) {
         return !this.isOptional() || (this.isActiveClosure ? this.isActiveClosure(script) : true)
     }
+
+    private static Set<JobType> JOB_TYPES = [
+        PULLREQUEST,
+        NIGHTLY,
+        // UPDATE_VERSION,
```
Comment (Ginxo) [inline dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/JobType.groovy]: to remove?

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy [hunk 6]
```diff
@@ -167,26 +84,169 @@ enum Folder {
...
+    }
+
     static List<Folder> getAllFolders(def script) {
         return getAllNightlyFolders(script) +
             getAllPullRequestFolders(script) +
             getAllReleaseFolders(script) +
             getAllToolsFolders(script) +
             getAllOtherFolders(script)
-    // Folder.UPDATE_VERSION,
+            // Folder.UPDATE_VERSION,
```
Comment (Ginxo) [inline dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy]: to remove? is this on purpose?

**Review by Ginxo — APPROVED** [review]
LGTM, just a proposal about how to do the same (I'm not against your solution) and just to pay attention on commented lines.
Congrats!

File: dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy (hunk 3 above)
Comment (radtriste) [reply dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl/model/Folder.groovy]: this is for future improvement. but I can remove for now

File: dsl/seed/jobs/Jenkinsfile.seed.branch (hunk 1 above)
Comment (radtriste) [reply dsl/seed/jobs/Jenkinsfile.seed.branch:102]: gonna update

Comment (radtriste) [reply dsl/seed/jobs/Jenkinsfile.seed.branch:102]: See my changes below

**Pull request merged**